File: feedback/src/main/java/com/kanetik/feedback/model/ContextDataItem.java
package com.kanetik.feedback.model;

import android.support.annotation.Keep;

import java.io.Serializable;

@Keep
public class ContextDataItem implements Serializable {
    static final long serialVersionUID = 326L;

    public String key;
    public Object value;

    public ContextDataItem(String key, Object value) {
        this.key = key;
        this.value = value;
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof ContextDataItem)) {
            return false;
        }

        ContextDataItem that = (ContextDataItem) other;
        return this.key.equals(that.key);
    }
}

File: feedback/src/main/java/com/kanetik/feedback/model/ContextDataItem.kt
package com.kanetik.feedback.model

import kotlinx.serialization.*

@Serializable
data class ContextDataItem(val key: String, val stringValue: String?, val intValue: Int?) {
    constructor(key: String, stringValue: String): this(key, stringValue, null)
    constructor(key: String, intValue: Int?): this(key, null, intValue)

    override fun equals(other: Any?): Boolean {
        if (other !is ContextDataItem) {
            return false
        }

        return key == other.key && (stringValue == other.stringValue || intValue == other.intValue)
    }

    fun getValue(): String {
        return stringValue ?: intValue.toString()
    }
}